pyfakefs/fake_filesystem.py

```diff
@@ -212,6 +218,7 @@ def __init__(
         self.path_separator: str = path_separator
         self.alternative_path_separator: Optional[str] = os.path.altsep
         self.patcher = patcher
+        self.create_temp_dir = create_temp_dir
```

Comment:
This is called twice in`__init__`, here and at line 257. Why is does the temp dir need to be created twice?